crates/dekaf/src/task_manager.rs

```diff
@@ -85,8 +89,21 @@ impl TaskStateListener {
             // Scope to force the borrow to end before awaiting
             {
                 let current_value = temp_rx.borrow_and_update();
-                if let Some(ref result) = *current_value {
-                    return result.clone().map_err(anyhow::Error::from);
+                match &*current_value {
```

Comment:
We used to happily hand out expired tokens here, just assuming that the `TaskManager` loop could keep up. That's the real cause of the problem